ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/NewWallet.hs

```diff
@@ -0,0 +1,235 @@
+  QBoxLayout.setSpacing layout 18
+  QLayout.setContentsMarginsRaw layout 24 42 24 24
+
+  walletNameLabel <- QLabel.newWithText ("<b>WALLET NAME</b>" :: String)
+  walletName <- QLineEdit.new
+  QLineEdit.setPlaceholderText walletName ("wallet name" :: String)
+
+  addRow layout walletNameLabel walletName
+  addSeparator layout
+
+  hasPasswordLabel <- QLabel.newWithText $ toString passwordLabelText
+  QLabel.setWordWrap hasPasswordLabel True
+  QWidget.setMinimumSizeRaw hasPasswordLabel 600 30
+  hasPassword <- QCheckBox.new
+  QAbstractButton.setChecked hasPassword True
+  QWidget.setSizePolicyRaw hasPassword Maximum Preferred
+
+  addRow layout hasPasswordLabel hasPassword
+
+  passwordWidget <- QWidget.new
+  passwordLayout <- QVBoxLayout.new
+  QWidget.setLayout passwordWidget passwordLayout
+  QLayout.setContentsMarginsRaw passwordLayout 0 0 0 0
+  QBoxLayout.setSpacing passwordLayout 18
+  QWidget.setSizePolicyRaw passwordWidget Preferred Minimum
+  QLayout.setSizeConstraint passwordLayout QLayout.SetMinimumSize
+
+  passwordLabel <- QLabel.newWithText ("<b>PASSWORD</b>" :: String)
+  (passwordFieldLayout, password) <- createPasswordField "password"
+
+  addRowLayout passwordLayout passwordLabel passwordFieldLayout
+  addSeparator passwordLayout
+
+  repeatPasswordLabel <- QLabel.newWithText ("<b>REPEAT PASSWORD</b>" :: String)
+  (repeatPasswordFieldLayout, repeatPassword) <- createPasswordField "password"
+
+  addRowLayout passwordLayout repeatPasswordLabel repeatPasswordFieldLayout
+
+  QBoxLayout.addWidget layout passwordWidget
+
+  QBoxLayout.addSpacing layout 24
+  createButton <- QPushButton.newWithText ("CREATE" :: String)
+  QObject.setObjectName createButton ("createButton" :: String)
+  QWidget.setSizePolicyRaw createButton Maximum Preferred
+  QBoxLayout.addWidget layout createButton
+  void $ QLayout.setWidgetAlignment layout createButton $ alignHCenter .|. alignVCenter
+
+  QBoxLayout.addStretch layout
+
+  let nw = NewWallet{..}
+
+  connect_ hasPassword QAbstractButton.toggledSignal $ hasPasswordToggled nw
+  connect_ walletName QLineEdit.textChangedSignal $ \_ -> revalidate nw
+  connect_ password QLineEdit.textChangedSignal $ \_ -> revalidate nw
+  connect_ repeatPassword QLineEdit.textChangedSignal $ \_ -> revalidate nw
+  connect_ createButton QAbstractButton.clickedSignal $ \_ -> QDialog.accept newWallet
+
+  revalidate nw
+
+  return nw
+
+runNewWallet:: IO NewWalletResult
+runNewWallet = do
+  nw@NewWallet{..} <- initNewWallet
+  result <- fmap toEnum . liftIO . QDialog.exec $ newWallet
+
+  case result of
+    QDialog.Accepted -> maybe NewWalletCanceled NewWalletAccepted <$> fillWaletParameters nw
+    QDialog.Rejected -> return NewWalletCanceled
+
+passwordLabelText :: Text
+passwordLabelText =
+  "Activate to create password. Note that good password should be at least <b>7 characters long</b>, and have at least <b>1 uppercase character</b>, <b>1 lowercase character</b> and <b>1 digit</b>."
```

Comment:
Even though it's a string literal, I would like line width to be at most 100 characters (better 80).
You can use `<>` or such syntax:
```
  "aaa\
  \bbb"